### ISSUE TITLE
Fortran 90 support checks

### ISSUE THREAD

**Issue by rondiplomatico** [OpenCMISS/manage#4]
For systems with no gfortran (g77 or the like) the configuration runs through but for obvious reasons the build fails. so we need include some logic at main level that ensures f90 capabilities are present or fail otherwise. cmake already does that check - we just need to see what variable is pre-set for f90 caps. as we dont need fortran for zinc this should also be made iron-dependent.

**Comment by chrispbradley** [comment]
At the moment there is no Fortran with Zinc but this does not mean there will not be. Fortran is one of the supported languages and so there is no reason why there should not be Fortran bindings for Zinc. What ever Fortran variables are set in CMake should really be global rather than component dependent.

**Comment by rondiplomatico** [comment]
jup as soon as zinc will need fortran-90 this must be made global. however, for now, requiring presence of (a f90-capable) fortran compiler at all is unnecessary if you only want to build zinc.
what other fortran variables do you refer to?

**Comment by rchristie** [comment]
Failure to get compiler, tools or dependencies particular to one component shouldn't prevent others from being built. I am happy with Daniel's call on this, and prefer not to over-generalise until it's clear it's required.

**Comment by chrispbradley** [comment]
No, I'm not saying that you must have an F90 compiler for zinc. Just saying that you don't need to make any cmake variables specific to a component. They are fine as global variables. In any case, they way other build systems deal with things like this is that they check for the F90 compiler. If it is present they build the Fortran bindings. If it is not present they configure it so that they don't get built. There is normally a command line or other config flag to force them to be built in which case the absence of a F90 compiler would be an error that halts the build process. Richard, Fortran is a supported language and so they are required. Just run a python script that sucks in the .h files and pushes out a .f90 file. No big deal??? This is done for FieldML (C->f90) and we do it for the C bindings (f90->C).

**Comment by rondiplomatico** [comment]
Nah, i did not intend to make this a component-based variable. The main script sorting out the components and dependencies knows that it needs to build iron (or any other component using fortran) or not, and if so, we can check that the compiler is f90-capable.
